packages/wallet/src/redux/__tests__/helpers.ts

```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
-    expect(item.messagePayload.type).toEqual(messageType);
-    expect(item.messagePayload.signedCommitment.commitment).toMatchObject(c);
-  });
-};
 
-export const expectThisMessageAndTheseCommitmentsSent = (
+type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
+
+function transformCommitmentToMatcher(sc: { commitment: Partial<Commitment>; signature?: string }) {
+  if (sc.signature) {
+    return expect.objectContaining({ signature: sc.signature });
+  } else {
+    return expect.objectContaining({ commitment: expect.objectContaining(sc.commitment) });
+  }
+}
+
+export const expectThisCommitmentSent = (
   state: SideEffectState,
-  commitments: PartialCommitments,
-  messageType: string,
+  commitment: Partial<Commitment>,
+  type = 'WALLET.COMMON.COMMITMENT_RECEIVED',
+  idx = 0,
 ) => {
-  expectSideEffect('messageOutbox', state, item => {
-    expect(item.messagePayload.type).toEqual(messageType);
-    expect(item.messagePayload.signedCommitments).toMatchObject(
-      // If the user passes a signature, we should match against it. Otherwise,
-      // the signature should be present, but we don't care what its value is
-      commitments.map(({ commitment, signature }) => ({
-        commitment,
-        signature: signature || expect.any(String),
-      })),
+  const messageOutbox = getOutboxState(state, 'messageOutbox');
+
+  try {
+    // Passes when at least one message matches
+    // In the case of multiple messages queued, this approach does not care about
+    // their order, which is beneficial.
+    // However, the diffs produced by jest are unreadable, so when this expectation fails,
+    // we catch the error below
+    expect(messageOutbox).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          messagePayload: expect.objectContaining({
+            signedCommitment: transformCommitmentToMatcher({ commitment }),
+          }),
+        }),
+      ]),
     );
-  });
-};
+  } catch (err) {
+    if ('matcherResult' in err) {
+      // In this case, we've caught a jest expectation error.
+      // We try to help the developer by using expect(foo).toMatchObject(bar)
+      // The errors are much more useful in this case, but will be deceiving in the case when
+      // multiple messages are queued.
+
+      // To help with debugging, you can change the idx variable when running tests to 'search'
+      // for the correct commitment
 
-export const expectThisCommitmentSent = (state: SideEffectState, c: Partial<Commitment>) => {
-  expectThisMessageAndCommitmentSent(state, c, 'WALLET.COMMON.COMMITMENT_RECEIVED');
+      expect(messageOutbox[idx]).toMatchObject({
+        messagePayload: {
+          type,
+          signedCommitment: { commitment },
+        },
+      });
+    } else {
+      throw err;
+    }
+  }
 };
-type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
 
 export const expectTheseCommitmentsSent = (
```

Comment:
Should this use `expectThisCommitmentSent`?

Reply:
No, because `expectThisCommitmentSent` is about sending `COMMITMENT_RECEIVED`, which only works for two-party channels.

In an n-party channel, we are currently sending the whole round of commitments, as this is the only thing that allows "forward-only" communication to work. This is a different action: `COMMITMENTS_RECEIVED`

Reply:
Gotcha. It's unfortunate that most of the functionality between the `expectThisCommitmentSent` and `expectTheseCommitmentsSent` is copy-pasted. But this is test code after all...

Reply:
Yes, I thought about trying to dry it out a bit, but deemed it not worth the effort.